clang/lib/DPCT/ASTTraversal.h

```diff
@@ -1323,16 +1323,30 @@ class MemVarRule : public NamedMigrationRule<MemVarRule> {
   void registerMatcher(ast_matchers::MatchFinder &MF) override;
   void runRule(const ast_matchers::MatchFinder::MatchResult &Result);
 
+private:
+  void processTypeDeclaredLocal(const VarDecl *MemVar,
+                                std::shared_ptr<MemVarInfo> Info);
+};
+
+class ConstantMemVarRule : public NamedMigrationRule<ConstantMemVarRule> {
+public:
+  void registerMatcher(ast_matchers::MatchFinder &MF) override;
+  void runRule(const ast_matchers::MatchFinder::MatchResult &Result);
+
 private:
   void previousHCurrentD(const VarDecl *VD, tooling::Replacement &R);
   void previousDCurrentH(const VarDecl *VD, tooling::Replacement &R);
   void removeHostConstantWarning(tooling::Replacement &R);
-  void processTypeDeclaredLocal(const VarDecl *MemVar,
-                                std::shared_ptr<MemVarInfo> Info);
   bool currentIsDevice(const VarDecl *MemVar, std::shared_ptr<MemVarInfo> Info);
   bool currentIsHost(const VarDecl *VD, std::string VarName);
 };
 
+class MemVarRefRule : public NamedMigrationRule<MemVarRefRule> {
```

Comment:
Please update the name of the rule, then reader can get out it is analysis or migration rule easily.